### PR TITLE
Add ArborX to travis

### PR DESCRIPTION
**Pull request by streeve**
Add canary stage and optional ArborX build.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,30 +38,43 @@ addons:
...
     make -j2 VERBOSE=1 && make install &&
     popd && popd
-  - pwd
+  - if [[ ${ArborX} ]];  then
+      git clone --depth=1 https://github.com/arborx/ArborX.git &&
+      pushd ArborX && mkdir build && pushd build &&
+      cmake -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
+            -DCMAKE_PREFIX_PATH=${HOME}/kokkos
+            -DCMAKE_INSTALL_PREFIX=${HOME}/ArborX
+            -DARBORX_ENABLE_MPI=ON
```
Comment (dalg24) [inline .travis.yml]: I don't see a problem with enabling MPI in ArborX, but just making sure you know you don't need it at this time.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -67,6 +67,10 @@ target_include_directories(CabanaMD PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}
 
 target_link_libraries(CabanaMD Cabana::cabanacore Kokkos::kokkos MPI::MPI_CXX dl hwloc)
 
+if(CabanaMD_ENABLE_ArborX)
+  target_link_libraries(CabanaMD ArborX::ArborX)
```
Comment (dalg24) [inline src/CMakeLists.txt]: I expect you would transitively get it from Cabana.

Comment (dalg24) [reply src/CMakeLists.txt]: Similarly, I don't think you need to specify the `Kokkos::kokkos` target above

Comment (junghans) [reply src/CMakeLists.txt]: Yeah, we should fix that in Cabana.

Comment (junghans) [reply src/CMakeLists.txt]: Yeah, we should fix that in Cabana.

**Comment by streeve** [comment]
Will rebase once #54 is done... and #55

**Comment by streeve** [comment]
@dalg24 can you re-review?

File: .travis.yml [hunk 3]
```diff
@@ -16,30 +38,42 @@ addons:
...
   - git clone --depth=1 https://github.com/ECP-copa/Cabana.git
   - pushd Cabana && mkdir build && pushd build &&
     cmake -DCMAKE_CXX_COMPILER_LAUNCHER=ccache
-          -DCMAKE_PREFIX_PATH=$HOME/kokkos
+          -DCMAKE_PREFIX_PATH="$HOME/kokkos;$HOME/ArborX"
           -DCMAKE_INSTALL_PREFIX=$HOME/Cabana
-          -DCabana_REQUIRE_SERIAL=OFF
           ${CABANA_OPTS[@]}
-          -DCabana_ENABLE_MPI=ON
+          -DCabana_ENABLE_MPI=ON -DCabana_ENABLE_ARBORX=${ArborX}
```
Comment (dalg24) [inline .travis.yml:73]: I was initially surprised that it works if `ArborX` is not defined but it seems to be ok

**Review by dalg24 — APPROVED** [review]

File: .travis.yml (hunk 3 above)
Comment (streeve) [reply .travis.yml:73]: I think that's the "default if found" behavior that was added

**Pull request merged**